### PR TITLE
Add inline query

### LINKED ISSUE #8: Add inline option?
I love this extension, I'd love to see an option where we can inline the type right beside the variable:

```ts
pizza; // ? const pizza: Pizza
```

This is really helpful when teaching and sharing code because you can make the types visible without taking up too much room

### PR DESCRIPTION
**Pull request by tommy-mitchell**
Resolves #8 

Adds an inline query. If a `=>`[^1] is present at the end of a line, displays the first inlay hint found, starting from the left of the line:

```ts
const x = { y: "z" };
x.y; //=> `const x: { y: string };`
```

I've refactored logic into `queries.ts` and `helpers.ts`.

[^1]: Maybe the `<?` query could display the right-most hint? A number after both could display the next hint in line.

### CODE REVIEW

File: src/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  // Make a one-liner
+  let text = hint.body.displayString
+    .replace(/\\n/g, " ")
+    .replace(/\/n/g, " ")
+    .replace(/  /g, " ")
+    .replace(/[\u0000-\u001F\u007F-\u009F]/g, "");
+  
+  // Cut off hint if too long
+  // If microsoft/vscode#174159 lands, can change to check that
+  const availableSpace = 120 - lineLength;
```
Comment (tommy-mitchell) [inline src/helpers.ts:42]: https://github.com/microsoft/vscode/issues/174159 proposes a `TextEditor.width` API option, which would add a convenient way to hide long hints based on viewport width.

File: src/queries.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  for (const match of m) {
+    if (match.index === undefined) {
+      break;
+    }
+
+    const end = match.index + match[0].length - 1;
+    // Add the start range for the inlay hint
+    const endPos = model.positionAt(end + offset);
+
+    if (cancel.isCancellationRequested) {
```
Comment (tommy-mitchell) [inline src/queries.ts:44]: Can the cancellation token change in the middle of the for loop? If not, this check could be moved to `getHintsFromQueries()`

File: src/queries.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      return [];
+    }
```
Comment (tommy-mitchell) [inline src/queries.ts:46]: Can the cancellation token change in the middle of the for loop? If not, this check could be moved to `getHintsFromQueries()`

File: src/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  if (text.length > availableSpace) {
+    text = text.slice(0, availableSpace - 1) + "...";
+  }
```
Comment (tommy-mitchell) [inline src/helpers.ts:45]: https://github.com/microsoft/vscode/issues/174159 proposes a `TextEditor.width` API option, which would add a convenient way to hide long hints based on viewport width.

File: src/queries.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+/** Strongly-typed RegExp groups (https://github.com/microsoft/TypeScript/issues/32098#issuecomment-1279645368) */
+type RegExpGroups<T extends string> = (
+  & IterableIterator<RegExpMatchArray>
+  & {
+    groups: { [name in T]: string } | { [key: string]: string };
+  }[]
+);
+
+const twoslashQueryRegex = /^\s*\/\/\s*\^\?/gm; // symbol: ^?
+const inlineQueryRegex = /^[^\S\r\n]*(?<start>\S).*\/\/\s*(?<end>=>)/gm; // symbol: =>
```
Comment (tommy-mitchell) [inline src/queries.ts:18]: https://regex101.com/r/6Jb8h2/1 - skips whitespace and ensures that at least one word character exists before the symbol

File: src/queries.ts (hunk 3 above)
Comment (orta) [reply src/queries.ts:48]: Yes, because of the async requests, it seems pretty reasonable to me 👍🏻

**Comment by orta** [comment]
Alright - thanks I'll give this a run through locally

**Pull request merged**